src/mainfest.rs

```diff
@@ -0,0 +1,24 @@
+use std::{fs::File, io::Read as _};
+
+use toml_edit::{DocumentMut, Item, Value};
+
+/// Get the contents of the manifest file.
+fn get_cargo_toml(folder_name: &str) -> anyhow::Result<DocumentMut> {
+    let mut file = File::open(format!("{folder_name}/Cargo.toml"))?;
+
+    let mut content = String::new();
+    file.read_to_string(&mut content)?;
+
+    Ok(content.parse()?)
+}
+
+/// Determine the name of the cargo package.
+pub(crate) fn package_name() -> anyhow::Result<String> {
```

Comment:
This can potentially be replaced by parsing the output of the [`cargo metadata` command](https://doc.rust-lang.org/cargo/commands/cargo-metadata.html)